### ISSUE TITLE
Return the Future wherever applicable.

### ISSUE THREAD

**Issue by AmitJoki** [bartektartanus/pref_dessert#1]
I found myself in a situation where I wanted to wait until `removeAll()` was done but the return type of it was void.

It would be intuitive to return the Future. I've fixed it in my local code and thought this is what others might experience as well so might as well include it in the official branch.

I can try to make a PR if necessary.

**Comment by bartektartanus** [comment]
Good idea. I will change all this remove and save methods into `Future<bool>` which will indicate success/failure. Is that ok for you? :)

**Comment by AmitJoki** [comment]
@bartektartanus that would be fantastic!
